.github/workflows/java-compatible.yml

```diff
@@ -1,22 +1,20 @@
 
 name: Build with different JDKs
 
-on:
-  push:
-    branches:
-      - main
+on: push
```

Comment:
You also seem to be firing on `push` on any branch, rather than only on the main branch. This is a nontrivial change not included in the PR description. Was this intentional?

Reply:
yeah ... one of the issues was that I didn't have that running on feature branches.